crates/bevy_sprite/src/collide_aabb.rs

```diff
@@ -55,3 +60,132 @@ pub fn collide(a_pos: Vec3, a_size: Vec2, b_pos: Vec3, b_size: Vec2) -> Option<C
         None
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    fn collide_two_rectangles(
+        // (x, y, size x, size y)
+        a: (f32, f32, f32, f32),
+        b: (f32, f32, f32, f32),
+    ) -> Option<Collision> {
+        collide(
+            Vec3::new(a.0, a.1, 0.),
+            Vec2::new(a.2, a.3),
+            Vec3::new(b.0, b.1, 0.),
+            Vec2::new(b.2, b.3),
+        )
+    }
+
+    #[test]
+    fn inside_collision() {
+        // Identical
+        #[rustfmt::skip]
+        let res = collide_two_rectangles(
+            (1., 1., 1., 1.),
+            (1., 1., 1., 1.),
+        ).expect("Collision expected");
+        assert_eq!(res, Collision::Inside);
+        // B inside A
+        #[rustfmt::skip]
+        let res = collide_two_rectangles(
+            (2., 2., 2., 2.),
+            (2., 2., 1., 1.),
+        ).expect("Collision expected");
+        assert_eq!(res, Collision::Inside);
+        // A inside B
+        #[rustfmt::skip]
+        let res = collide_two_rectangles(
+            (2., 2., 1., 1.),
+            (2., 2., 2., 2.),
+        ).expect("Collision expected");
+        assert_eq!(res, Collision::Inside);
+    }
+
+    #[test]
+    fn collision_based_on_b() {
+        // Right of B
+        #[rustfmt::skip]
+        let res = collide_two_rectangles(
+            (3., 2., 2., 2.),
+            (2., 2., 2., 2.),
+        ).expect("Collision expected");
+        assert_eq!(res, Collision::Right);
+        // Left of B
+        #[rustfmt::skip]
+        let res = collide_two_rectangles(
+            (1., 2., 2., 2.),
+            (2., 2., 2., 2.),
+        ).expect("Collision expected");
+        assert_eq!(res, Collision::Left);
+        // Top of B
+        #[rustfmt::skip]
+        let res = collide_two_rectangles(
+            (2., 3., 2., 2.),
+            (2., 2., 2., 2.),
+        ).expect("Collision expected");
+        assert_eq!(res, Collision::Top);
+        // Bottom of B
+        #[rustfmt::skip]
+        let res = collide_two_rectangles(
+            (2., 1., 2., 2.),
+            (2., 2., 2., 2.),
+        ).expect("Collision expected");
+        assert_eq!(res, Collision::Bottom);
+    }
+
+    // In case the X-collision depth is equal to the Y-collision depth, always
+    // prefer X-collision, meaning, `Left` or `Right` over `Top` and `Bottom`.
```

Comment:
This is non-blocking, but I feel like rather than "locking" the implementation in a test, the behavior itself could be improved. It's not clear we do not want corner variants to the `Collision` enum.

Reply:
I don't see the tests as "locking" the function behavior. Instead, to me, it is verification for the documentation that I added plus learning material - a test is usually the first user.
I agree that one could debate the behavior of the function or its API (taking a `Vec3` for comparing rectangles) but this isn't the intention of this PR.